torch/csrc/jit/backends/backend_detail.cpp

```diff
@@ -124,6 +131,44 @@ Module codegen_backend_module(
             )",
       loweredModuleResolver());
 
+  // Debug info generated by BackendModuleDebugInfoRecorder
+  // WenNeed to have its lifetime tied to the module. It can either be in
+  // 1) Module or 2) In the TS module
+  // 1 requires us to introduce debug info that is backend specific into
+  // the Module class.
+  // So we opt for 2 at the moment. This is done as follows.
+  // We have a custom class BackendDebugInfo that stores debug info.
+  // This class is registered in lowered module and have a method
+  // that creates an instance of this class.
+  // We extract the instance and store debug info in it.
+  // This will be extracted at model saving time.
+  // This information will not be serialized and thus BackendDebugInfo.
+  // However a valid instance of BackendDebugInfo must exists when model is
+  // loaded. For example, say we embed a lowered module inside another python
+  // module. m = SomeModule(lowered_module) and then we do scripted_module =
+  // torch.jit.script(m) This step would fail, if valid instance of
+  // BackendDebugInfo existed in the model being scripted/serialized but did not
+  // exist when it was deserialized. Thus we create an instanace of
+  // BackendDebugInfo when backend is created in setstate. This instance will be
+  // just a dummy instance for lite interpreter (C10_MOBILE), which wont have
+  // same CPP class backing it. Unfortunately, these hacks are necessary in the
+  // absence of 1.
+  c10::intrusive_ptr<torch::CustomClassHolder> backend_debug_info_class;
+  const c10::QualifiedName backend_debug_info_class_name(
+      {"__torch__", "torch", "classes", "backend", "BackendDebugInfo"});
+  auto debug_info_cls =
+      getCustomClass(backend_debug_info_class_name.qualifiedName());
```

Comment:
`getCustomClass` can return `nullptr`, might be a good idea to add an assert here.